edit/linter-report.js

```diff
@@ -0,0 +1,161 @@
+/* global linter editors clipString createLinterHelpDialog makeSectionVisible */
+'use strict';
+
+var linterReport = (() => { // eslint-disable-line no-var
+  const cms = new Map();
+  const helpDialog = createLinterHelpDialog(getIssues);
+
+  linter.onChange((annotationsNotSorted, annotations, cm) => {
+    let table = cms.get(cm);
+    if (!table) {
+      table = createTable(cm);
+      cms.set(cm, table);
+      const container = $('.lint-report-container');
+      if (typeof editor === 'object') {
+        container.append(table.element);
+      } else {
+        const nextSibling = findNextSibling(cms, cm);
+        container.insertBefore(table.element, nextSibling && cms.get(nextSibling).element);
+      }
+    }
+    table.update();
+    table.updateAnnotations(annotations);
+    updateCount();
+  });
+
+  linter.onUnhook(cm => {
+    const table = cms.get(cm);
+    if (table) {
+      table.element.remove();
+      cms.delete(cm);
+    }
+    updateCount();
+  });
+
+  document.addEventListener('DOMContentLoaded', () => {
+    $('#lint-help').addEventListener('click', helpDialog.show);
+  }, {once: true});
+
+  return {refresh};
+
+  function updateCount() {
+    const issueCount = Array.from(cms.values()).reduce((sum, t) => {
```

Comment:
* sum is a local variable so no need to update it
* an array element should relate to the array name (`cm` here) or have a typical name like el or _ or item, etc.

`reduce((sum, cm) => sum + cm.trs.length, 0);`